packages/super-components/components/msc-tail-logs/src/msc-tail-logs/msc-tail-logs.tsx

```diff
@@ -0,0 +1,123 @@
+import {
+  Component,
+  Prop,
+  Element,
+  Host,
+  h,
+  State,
+  Watch,
+} from '@stencil/core';
+import { HTMLStencilElement } from '@stencil/core/internal';
+import { apiClient } from '@ovh-ux/manager-core-api';
+import './msc-tail-logs-code';
+
+const LOGS_REFRESH_INTERVAL = 30000;
+const LOGS_LIMIT = 500;
+
+enum SortOrder {
+  Ascending = 'asc',
+  Descending = 'desc',
+}
+
+export interface LogEntry {
+  [key: string]: any;
+}
+
+@Component({
+  tag: 'msc-tail-logs',
+  styleUrl: 'msc-tail-logs.scss',
+  shadow: true,
+})
+export class MscTailLogs {
+  @Prop() serviceName?: string = '';
+
+  @Prop() source?: string = '';
+
+  @Prop() sort?: SortOrder = SortOrder.Ascending;
+
+  @Prop() isToggled?: boolean = false;
+
+  @Prop() logsRefresh?: number = LOGS_REFRESH_INTERVAL;
+
+  @Prop() logLimit = LOGS_LIMIT;
+
+  @Element() host!: HTMLStencilElement;
+
+  @State() private logs: LogEntry[] = [];
+
+  private logsRefreshInterval: any = null;
+
+  async componentWillLoad() {
+    await this.fetchLogs(); // Fetch logs when the component loads
+  }
+
+  componentDidLoad() {
+    // Start the auto-refresh interval only if isToggled is true
+    if (this.isToggled) {
+      this.startLogsRefreshInterval();
+    }
+  }
+
+  @Watch('isToggled')
+
+  toggleChanged(newValue: boolean) {
+    if (newValue) {
+      this.startLogsRefreshInterval();
+    } else {
+      this.stopLogsRefreshInterval();
+    }
+  }
+
+  private async fetchLogs() {
```

Comment:
you can move this fetch inside msc-tail-logs-code.tsx and and fetch inside the file